scipio/src/dma_file.rs

```diff
@@ -70,6 +71,10 @@ fn align_down(v: u64, align: u64) -> u64 {
     v & !(align - 1)
 }
 
+pub trait AsSourceFd {
```

Comment:
I don't quite like this direction.

RawFd is very central to rust and used everywhere.

It is probably better to encode this information in the `SourceType`. We do that already for other things in the system and would have the added benefit of consistency.

See for instance how we can encode a buffer inside the SourceType for DmaRead.

Another option that would be acceptable would be to use a special `SourceType`. So we would have for instance `DmaRead` and `PollableDmaRead` Both directions would be acceptable so long as we keep RawFd untouched